nntrainer/tensor/tensor.cpp

```diff
@@ -720,6 +739,9 @@ Tensor Tensor::getSharedDataTensor(const TensorDim dim_, size_t offset,
                                    bool reset_stride,
                                    const std::string &name_) const {
   Tensor ret = *this;
+  if (dim_.getFormat() != ret.dim.getFormat())
```

Comment:
How about we making a function or macro for this? Checking Tensor Format will be needed in every tensor operation.
Or at least matching the error message with line 521 might be needed.